smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/DeconflictingStrategy.java

```diff
@@ -49,12 +49,12 @@
     private final Map<ShapeId, String> pointers = new HashMap<>();
     private final Map<String, ShapeId> reversePointers = new HashMap<>();
 
-    DeconflictingStrategy(Model model, RefStrategy delegate) {
+    DeconflictingStrategy(Model model, RefStrategy delegate, boolean preventUnitConflicts) {
         this.delegate = delegate;
 
         // Pre-compute a map of all converted shape refs. Sort the shapes
         // to make the result deterministic.
-        model.shapes().filter(FunctionalUtils.not(this::isIgnoredShape)).sorted().forEach(shape -> {
+        model.shapes().filter(s -> !isIgnoredShape(s, preventUnitConflicts)).sorted().forEach(shape -> {
```

Comment:
Instead of the boolean flag, could you iterate over every tagged union shape, then check if any of its members target smithy.api#Union? If that's the case, then include Union, otherwise, don't include it. Should be pretty fast since it would only need to check union shape members and that kind of filtering is cached in Model.

Reply:
Ah I see that's how you're determining the boolean flag. Nevermind.